script/update-build-tomcat.sh

```diff
@@ -7,66 +7,106 @@
 ######################################################
 
 HOST=$(uname -n)
-APP_MODE="calcentral"
+MYSCRIPT=`basename "$0"| sed "s/\..*//g"`
 
-# Setting tomcat latest build if no WAR_URL defined
-# Retire Torquebox - CalCentral Tomcat new branch
-WAR_URL=${WAR_URL:="https://bamboo-sis.berkeley.edu/browse/CAL-RTCW/latest/artifact/shared/warfile/calcentral.war"}
+# If the WAR_URL is not defined in the environment the deploy will abort for safety, we do not want to default to any branch
+WAR_URL=${WAR_URL}
 MAX_ASSET_AGE_IN_DAYS=${MAX_ASSET_AGE_IN_DAYS:="45"}
-DOC_ROOT="/var/www/html/${APP_MODE}"
+DOC_ROOT="/var/www/html/calcentral"
 
 CC_LOG_DIR="$HOME/calcentral/log"
 LOG=$(date +"${CC_LOG_DIR}/update-build-tomcat_%Y-%m-%d.log")
 LOGIT="tee -a ${LOG}"
 
-## cd $( dirname "${BASH_SOURCE[0]}" )/..
-# The assets will not be anymore under the calcentral checked out code
-# they are brought in the WAR file so changing directories to the correct location
-#cd ${TOMCAT_DEPLOY} || exit 1
+
+
+function show_help {
+
+    cat <<EOF
+
+CalCentral Code Deploy script usage:
+------------------------------------
+
+By default, it deploys code without a server restart
+
+   $ ./${MYSCRIPT}
+
+To Deploy offline with a server restart:
+----------------------------------------
+
+   $ ./${MYSCRIPT} -o offline
+
+     Note: The parameter needs to be written exactly as shown in the help above, any other keyword will not work.
+
+EOF
+
+}
+
+
+while getopts o:h option ; do
+    case $option in
+        o) OFFLINE=$OPTARG
+           if [ "${OFFLINE}" != "offline" ] ; then
+           echo "Offline parameter = ${OFFLINE}"
+           show_help
+           fi
+           ;;
+        *) show_help
+           exit 1 ;;
+    esac
+done
 
 # Enable rvm and use the correct Ruby version and gem set.
 [[ -s "${HOME}/.rvm/scripts/rvm" ]] && . "${HOME}/.rvm/scripts/rvm"
 source .rvmrc
 
+# If the WAR_URL is not defined in the environment the deploy will abort for safety, we do not want to default to any branch
+if [ -z "$WAR_URL" ] ; then
+     echo "WAR_URL variable not defined in the node"| ${LOGIT}
+     exit 1
+fi
+
 # Update source tree (from which these scripts run)
 
 # Change dir to the calcentral home dir to update the source code from GitHub
 cd ${CLC_HOME} || exit 1
-./script/update-source.sh
+#####./script/update-source.sh
```

Comment:
Does this line need to fully removed from this script?

Reply:
Yes, the line can be removed. I left it while we were testing.